### ISSUE TITLE
Broken RTL languages support.

### ISSUE THREAD

**Issue by shetozz** [jzohrab/lute#67]
**Description**

RTL display is broken, it's not completely LTR either, I'll provide Arabic text.
Without diacritics (which represent short vowels), it shows words correctly, but in the wrong order LTR instead of RTL, but when I add the diacritics, it loses its ability to show connected letters mostly.

Another point is, Titles show completely fine, with the diacritics and in the correct order.

**Reproduce**

[صَبَاحُ الخَير.md](https://github.com/jzohrab/lute/files/12676808/default.md)

 This is the file if you wanna try it on your system.

Edit: text from the file ... SORT OF ... even copying and pasting it here makes it behave strangely.  The exclamation marks belong on the left of the first two lines.  :-)

```
خالِد: صَبَاحُ الخَيرِ!
خُلُود: صَبَاحُ النُّور!
خالِد: كَيفَ حَالُكِ؟
خُلُود: بِخَيرٍ، وَأَنْتَ؟
خَالِد: فِي أَحْسَنِ حالٍ. ما اسمُكِ؟
خُلُود: اسمي خُلُودُ، وأَنتَ، ما اسمُكَ؟
خَالِد: أَنا خَالِد، تَشَرَّفْتُ بِلِقَائِكِ.
خُلُود: الشَّرَفُ لِي.
```

Screenshot of the text file:

<img width="317" alt="image" src="https://github.com/jzohrab/lute/assets/1637133/86f21d4c-53e9-4898-b6db-37fa60c7d0a0">

**Screenshots**
![Expected behavior](https://github.com/jzohrab/lute/assets/93838524/f74105fe-66d3-4b68-9461-5e73602ce862)
![With diacritics (AKA short vowels)](https://github.com/jzohrab/lute/assets/93838524/2aac68c1-97cd-4ef1-a62f-e37bf951616c)
![Without diacritics](https://github.com/jzohrab/lute/assets/93838524/f6f4d658-0215-4fed-a1e3-07c18b03e9f9)

**Extra software info, if not already included in the Description:**

- OS: Fedora 38 workstation edition (Gnome)
- Browser: fire fox.
- Web server: Idk, I'm running it locally, with a docker installation.
- Version: 2.0.14

**Comment by shetozz** [comment]
Adding letters with diacritics in the character field, and it solved one of the problems, which is the inability to connect letters when diacritics are present, now, we have the other one which is the wrong word order.

**Comment by jzohrab** [comment]
Thanks @shetozz for the info, I'll take a quick look to start to get a handle on the issue.

**Comment by jzohrab** [comment]
To explore, I've created a branch, `hack_issue_67_RTL_text` and started making some hacky changes to it, just to get an idea of what's involved.

By hardcoding `dir="rtl"` in a few places, I have the following:

> <img width="658" alt="image" src="https://github.com/jzohrab/lute/assets/1637133/0da8a450-936d-40a3-a809-325875b13279">

I haven't looked into the language word regex much, so you'll see that the "?" at the end of some sentences is included as part of the term.  Also note that the exclamation mark is on the wrong end of the sentence.

I don't know if the Term form (top right form) needs to do the text as right-to-left for the text boxes ... maybe it's good enough for just the text itself to be right to left, and all the text input boxes could be as-is.  (I understand that this is biased towards Western/Roman languages :-) I don't know how to easily switch all the layouts around!)  Also, the term definitions aren't right-to-left, I'm assuming that they're entering the defs in English or similar.  Lots of variables, I'm not sure how to work it all out.

A sample pop-up:

> <img width="203" alt="image" src="https://github.com/jzohrab/lute/assets/1637133/bf9c07b4-b603-4b8f-851a-68074d18fa8c">

I know the terms aren't related, just wanted to see what would happen.

Also note the text editing is still left-aligned:

> <img width="262" alt="image" src="https://github.com/jzohrab/lute/assets/1637133/86b1f759-9894-4b2d-8ee0-d77bdcb8e13f">

Wow this language looks tricky.

In short, perhaps just changing the layout of the text itself would be good enough ... not perfect, but good enough.  I don't know, haven't studied such a language, and so am flying blind!

**Comment by jzohrab** [comment]
@shetozz - if all we have to do to make this _somewhat functional_ (as a first iteration) is add `dir="rtl"` to a page, that's quite quick ... but I don't know what the implications would be.

Would you be willing to try out a special build, work with it for a bit, and see what problems you run into?  I don't know enough of the language to know if what I'm doing is right or not.  We could use a special release, like `lute-rtl-beta-v1` or something, and you could install it the same way you install Lute regularly, try it out for a while with some stories or something, and see how it goes.  My turnaround time for releasing fixes might not be great ... but who knows, maybe it's simple.  I looked at how LWT (Learning with Texts) handles it, and it seems like the impact is minimal.

**Comment by shetozz** [comment]
I'll be more than happy to try it out and test it, just keep in mind that I'm a noob when it comes to programming, I'll install it with some instructions.
As for the fix itself, I think having an RTL text is enough for now, for the definitions, we can have a switch or something in the future to give the user choice to use English or Arabic for definitions.
But for now, I'll try the RTL text field.

**Comment by jzohrab** [comment]
No problem, it will be a regular install just like you do right now.  I'll only do the RTL text itself, everything else will be left-to-right.

**Comment by jzohrab** [comment]
Hi @shetozz, I've created a new "beta" release for RTL language support.

https://github.com/jzohrab/lute/releases/tag/rtl-beta-v0.1

You can delete your existing Lute installation (assuming you haven't actually used it for anything!!) and install the beta release just like you did with the prior release.  This beta release pre-defines Arabic and a short Arabic sample story (along with all the other languages).

Ping me on Discord with any questions you might have.  **New discord channel: #right-to-left-languages**.  Thanks! Jeff

**Comment by jzohrab** [comment]
Good enough to be merged into the dev branch, at least as a first release.

**Comment by jzohrab** [comment]
Was launched in 2.1.1. :tada:
